details.js

```diff
@@ -0,0 +1,79 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const urlParams = new URLSearchParams(window.location.search);
+  const countryName = urlParams.get('country');
+
+  if (!countryName) {
+    document.getElementById('country-name').innerHTML = '<p>No country selected!</p>';
+    return;
+  }
+
+  fetch('CountriesData.json')
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Failed to load JSON data');
+      }
+      return response.json();
+    })
+    .then(countries => {
+      const country = countries.find(c => c.name.toLowerCase() === countryName.toLowerCase());
+
+      //If the country is found, display its details
+      if (country) {
+        // Create the main container
```

Comment:
Please avoid explaining the code with comments, is a bad practice. 
The code should explain itself